lib/ssh_scan/api.rb

```diff
@@ -103,10 +104,23 @@ class API < Sinatra::Base
         options[:sockets] <<
           "#{params[:target]}:#{params[:port] ? params[:port] : "22"}"
         options[:policy_file] = options[:policy]
+        options[:force] = params[:force] ? params[:force] : false
+        unless (options[:force] == 'true')
+          available_result = settings.db.fetch_available_result(params)
+          unless available_result.nil?
+            if ((Time.now - Time.parse(available_result[:scanned_on].to_s))/(60*60*24) < 1)
+              return {
+                uuid: available_result[:uuid],
+                scanned_on: available_result[:scanned_on]
+              }.to_json
+            end
+          end
+        end
         options[:uuid] = SecureRandom.uuid
         settings.job_queue.add(options)
         {
-          uuid: options[:uuid]
+          uuid: options[:uuid],
+          scanned_on: 'fresh scan'
```

Comment:
I think there are two places we can handle this...

1.) We need a way to determine if a target is waiting in the JobQueue (there could be a method added to that object to check and if so, return the uuid to the end user)
2.) We need a way to determine the target has a recent scan in the Database. (there could be a method added to SSHScan::Database to check for pre-existing and recent scans from the target and return the UUID for that scan if it's within the cache timeout window.  This would also require respective interfaces added to MongoDB and SQLite objects that appear to operate the same from the call point of view.  Let me  know if you'd like some guidance here on that part.